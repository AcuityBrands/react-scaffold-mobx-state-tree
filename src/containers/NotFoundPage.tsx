import * as React from "react";
import { Link } from 'react-router-dom';
import { Button, Row, Col } from 'antd';

export default class NotFoundPage extends React.Component {
  render() {
    return (
      <div className="content-wrapper">
        <Row type="flex" justify="center">
          <Col sm={24} md={17} lg={12}>
            <div className="card">
              <h1>404</h1>
              <h3 className="mt10">Whatever you were trying to do, it didn't work</h3>
              <p className="mt20">
                Once upon a midnight dreary, While I websurfed, weak and weary,
                Over many a strange and spurious website of data galore,
              </p>
              <p className="mt20">
                While I clicked my fav'rite bookmark, Suddenly there came a warning,
                And my heart was filled with mourning, Mourning for my dear amour.
              </p>
              <p className="mt20">
                'Tis not possible!, I pleaded, But my browser, so conceited, Remained
                blank, I then repeated, Just a blank and nothing more.
              </p>
              <p className="mt20">
                With a scream, I was defeated, For my cookies were deleted, So i
                begged, no longer seated, "Give me back my charts hardcore!"
              </p>
              <p className="mt20">
                Then, in answer to my query, Through the net I loved so dearly,
                Came its answer, dark and dreary: Quoth the server, 404
              </p>
              <p className="mt20">
                <Button><Link to="/">Take me back to where it's warm and safe</Link></Button>
                {/* <Link to="/">.</Link> */}
              </p>
            </div>
          </Col>
          <Col sm={24} md={5} lg={5} offset={1}>
            <h3 className="mt20">About This Page</h3>
            <p className="mt10 light">
              This is a catch-all page.  Go ahead, try throwing a random url into the browser
              and you'll be brought right back here.  BOOM!
            </p>
            <p className="mt10 light">
              Just in case you have a hard time navigating your way around, here's a poem 
              to brighten your mood.
            </p>
          </Col>
        </Row>
      </div>
    );
  }
}