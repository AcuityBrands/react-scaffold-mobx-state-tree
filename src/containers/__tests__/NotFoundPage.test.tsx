import * as React from "react";
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import NotFoundPage from '../NotFoundPage';

Enzyme.configure({ adapter: new Adapter() });

const setup = propOverrides => {
  const props = Object.assign({}, propOverrides)
  const wrapper = Enzyme.shallow(<NotFoundPage {...props} />)
  return {
    props,
    wrapper
  }
}

describe('NotFoundPage Component', () => {
  // Minimal component test confirms component rendered
  it("can render", () => {
    const { wrapper } = setup({});
    expect(wrapper.exists()).toBe(true)
  })
})