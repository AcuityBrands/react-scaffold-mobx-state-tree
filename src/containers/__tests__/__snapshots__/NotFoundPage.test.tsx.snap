// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage Component matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFoundPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Row
        align="middle"
        gutter={0}
        justify="center"
        type="flex"
      >
        <Col
          lg={16}
          md={18}
          sm={20}
          xs={23}
        >
          <div
            className="card"
          >
            <h1>
              Oh, Hello 404
            </h1>
            <h3
              className="mt10"
            >
              Whatever you were trying to do, it didn't work
            </h3>
            <p
              className="mt20"
            >
              Once upon a midnight dreary, While I websurfed, weak and weary, Over many a strange and spurious websites or blogs galore,
            </p>
            <p
              className="mt20"
            >
              While I clicked my fav'rite bookmark, Suddenly there came a warning, And my heart was filled with mourning, Mourning for my dear amour.
            </p>
            <p
              className="mt20"
            >
              'Tis not possible!, I pleaded, But my browser, so conceited, Remained blank, I then repeated, Just a blank and nothing more.
            </p>
            <p
              className="mt20"
            >
              With a scream, I was defeated, For my cookies were deleted, So i begged, no longer seated, "Give me back my hrefs once more!"
            </p>
            <p
              className="mt20"
            >
              Then, in answer to my query, Through the net I loved so dearly, Came its answer, dark and dreary: Quoth the server, 404
            </p>
          </div>
        </Col>
      </Row>,
      "style": Object {
        "alignItems": "center",
        "background": "#f7f7f7",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "middle",
        "children": <Col
          lg={16}
          md={18}
          sm={20}
          xs={23}
        >
          <div
            className="card"
          >
            <h1>
              Oh, Hello 404
            </h1>
            <h3
              className="mt10"
            >
              Whatever you were trying to do, it didn't work
            </h3>
            <p
              className="mt20"
            >
              Once upon a midnight dreary, While I websurfed, weak and weary, Over many a strange and spurious websites or blogs galore,
            </p>
            <p
              className="mt20"
            >
              While I clicked my fav'rite bookmark, Suddenly there came a warning, And my heart was filled with mourning, Mourning for my dear amour.
            </p>
            <p
              className="mt20"
            >
              'Tis not possible!, I pleaded, But my browser, so conceited, Remained blank, I then repeated, Just a blank and nothing more.
            </p>
            <p
              className="mt20"
            >
              With a scream, I was defeated, For my cookies were deleted, So i begged, no longer seated, "Give me back my hrefs once more!"
            </p>
            <p
              className="mt20"
            >
              Then, in answer to my query, Through the net I loved so dearly, Came its answer, dark and dreary: Quoth the server, 404
            </p>
          </div>
        </Col>,
        "gutter": 0,
        "justify": "center",
        "type": "flex",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="card"
          >
            <h1>
              Oh, Hello 404
            </h1>
            <h3
              className="mt10"
            >
              Whatever you were trying to do, it didn't work
            </h3>
            <p
              className="mt20"
            >
              Once upon a midnight dreary, While I websurfed, weak and weary, Over many a strange and spurious websites or blogs galore,
            </p>
            <p
              className="mt20"
            >
              While I clicked my fav'rite bookmark, Suddenly there came a warning, And my heart was filled with mourning, Mourning for my dear amour.
            </p>
            <p
              className="mt20"
            >
              'Tis not possible!, I pleaded, But my browser, so conceited, Remained blank, I then repeated, Just a blank and nothing more.
            </p>
            <p
              className="mt20"
            >
              With a scream, I was defeated, For my cookies were deleted, So i begged, no longer seated, "Give me back my hrefs once more!"
            </p>
            <p
              className="mt20"
            >
              Then, in answer to my query, Through the net I loved so dearly, Came its answer, dark and dreary: Quoth the server, 404
            </p>
          </div>,
          "lg": 16,
          "md": 18,
          "sm": 20,
          "xs": 23,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Oh, Hello 404
              </h1>,
              <h3
                className="mt10"
              >
                Whatever you were trying to do, it didn't work
              </h3>,
              <p
                className="mt20"
              >
                Once upon a midnight dreary, While I websurfed, weak and weary, Over many a strange and spurious websites or blogs galore,
              </p>,
              <p
                className="mt20"
              >
                While I clicked my fav'rite bookmark, Suddenly there came a warning, And my heart was filled with mourning, Mourning for my dear amour.
              </p>,
              <p
                className="mt20"
              >
                'Tis not possible!, I pleaded, But my browser, so conceited, Remained blank, I then repeated, Just a blank and nothing more.
              </p>,
              <p
                className="mt20"
              >
                With a scream, I was defeated, For my cookies were deleted, So i begged, no longer seated, "Give me back my hrefs once more!"
              </p>,
              <p
                className="mt20"
              >
                Then, in answer to my query, Through the net I loved so dearly, Came its answer, dark and dreary: Quoth the server, 404
              </p>,
            ],
            "className": "card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Oh, Hello 404",
              },
              "ref": null,
              "rendered": "Oh, Hello 404",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Whatever you were trying to do, it didn't work",
                "className": "mt10",
              },
              "ref": null,
              "rendered": "Whatever you were trying to do, it didn't work",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Once upon a midnight dreary, While I websurfed, weak and weary, Over many a strange and spurious websites or blogs galore,",
                "className": "mt20",
              },
              "ref": null,
              "rendered": "Once upon a midnight dreary, While I websurfed, weak and weary, Over many a strange and spurious websites or blogs galore,",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "While I clicked my fav'rite bookmark, Suddenly there came a warning, And my heart was filled with mourning, Mourning for my dear amour.",
                "className": "mt20",
              },
              "ref": null,
              "rendered": "While I clicked my fav'rite bookmark, Suddenly there came a warning, And my heart was filled with mourning, Mourning for my dear amour.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "'Tis not possible!, I pleaded, But my browser, so conceited, Remained blank, I then repeated, Just a blank and nothing more.",
                "className": "mt20",
              },
              "ref": null,
              "rendered": "'Tis not possible!, I pleaded, But my browser, so conceited, Remained blank, I then repeated, Just a blank and nothing more.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "With a scream, I was defeated, For my cookies were deleted, So i begged, no longer seated, \\"Give me back my hrefs once more!\\"",
                "className": "mt20",
              },
              "ref": null,
              "rendered": "With a scream, I was defeated, For my cookies were deleted, So i begged, no longer seated, \\"Give me back my hrefs once more!\\"",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Then, in answer to my query, Through the net I loved so dearly, Came its answer, dark and dreary: Quoth the server, 404",
                "className": "mt20",
              },
              "ref": null,
              "rendered": "Then, in answer to my query, Through the net I loved so dearly, Came its answer, dark and dreary: Quoth the server, 404",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Row
          align="middle"
          gutter={0}
          justify="center"
          type="flex"
        >
          <Col
            lg={16}
            md={18}
            sm={20}
            xs={23}
          >
            <div
              className="card"
            >
              <h1>
                Oh, Hello 404
              </h1>
              <h3
                className="mt10"
              >
                Whatever you were trying to do, it didn't work
              </h3>
              <p
                className="mt20"
              >
                Once upon a midnight dreary, While I websurfed, weak and weary, Over many a strange and spurious websites or blogs galore,
              </p>
              <p
                className="mt20"
              >
                While I clicked my fav'rite bookmark, Suddenly there came a warning, And my heart was filled with mourning, Mourning for my dear amour.
              </p>
              <p
                className="mt20"
              >
                'Tis not possible!, I pleaded, But my browser, so conceited, Remained blank, I then repeated, Just a blank and nothing more.
              </p>
              <p
                className="mt20"
              >
                With a scream, I was defeated, For my cookies were deleted, So i begged, no longer seated, "Give me back my hrefs once more!"
              </p>
              <p
                className="mt20"
              >
                Then, in answer to my query, Through the net I loved so dearly, Came its answer, dark and dreary: Quoth the server, 404
              </p>
            </div>
          </Col>
        </Row>,
        "style": Object {
          "alignItems": "center",
          "background": "#f7f7f7",
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "middle",
          "children": <Col
            lg={16}
            md={18}
            sm={20}
            xs={23}
          >
            <div
              className="card"
            >
              <h1>
                Oh, Hello 404
              </h1>
              <h3
                className="mt10"
              >
                Whatever you were trying to do, it didn't work
              </h3>
              <p
                className="mt20"
              >
                Once upon a midnight dreary, While I websurfed, weak and weary, Over many a strange and spurious websites or blogs galore,
              </p>
              <p
                className="mt20"
              >
                While I clicked my fav'rite bookmark, Suddenly there came a warning, And my heart was filled with mourning, Mourning for my dear amour.
              </p>
              <p
                className="mt20"
              >
                'Tis not possible!, I pleaded, But my browser, so conceited, Remained blank, I then repeated, Just a blank and nothing more.
              </p>
              <p
                className="mt20"
              >
                With a scream, I was defeated, For my cookies were deleted, So i begged, no longer seated, "Give me back my hrefs once more!"
              </p>
              <p
                className="mt20"
              >
                Then, in answer to my query, Through the net I loved so dearly, Came its answer, dark and dreary: Quoth the server, 404
              </p>
            </div>
          </Col>,
          "gutter": 0,
          "justify": "center",
          "type": "flex",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="card"
            >
              <h1>
                Oh, Hello 404
              </h1>
              <h3
                className="mt10"
              >
                Whatever you were trying to do, it didn't work
              </h3>
              <p
                className="mt20"
              >
                Once upon a midnight dreary, While I websurfed, weak and weary, Over many a strange and spurious websites or blogs galore,
              </p>
              <p
                className="mt20"
              >
                While I clicked my fav'rite bookmark, Suddenly there came a warning, And my heart was filled with mourning, Mourning for my dear amour.
              </p>
              <p
                className="mt20"
              >
                'Tis not possible!, I pleaded, But my browser, so conceited, Remained blank, I then repeated, Just a blank and nothing more.
              </p>
              <p
                className="mt20"
              >
                With a scream, I was defeated, For my cookies were deleted, So i begged, no longer seated, "Give me back my hrefs once more!"
              </p>
              <p
                className="mt20"
              >
                Then, in answer to my query, Through the net I loved so dearly, Came its answer, dark and dreary: Quoth the server, 404
              </p>
            </div>,
            "lg": 16,
            "md": 18,
            "sm": 20,
            "xs": 23,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Oh, Hello 404
                </h1>,
                <h3
                  className="mt10"
                >
                  Whatever you were trying to do, it didn't work
                </h3>,
                <p
                  className="mt20"
                >
                  Once upon a midnight dreary, While I websurfed, weak and weary, Over many a strange and spurious websites or blogs galore,
                </p>,
                <p
                  className="mt20"
                >
                  While I clicked my fav'rite bookmark, Suddenly there came a warning, And my heart was filled with mourning, Mourning for my dear amour.
                </p>,
                <p
                  className="mt20"
                >
                  'Tis not possible!, I pleaded, But my browser, so conceited, Remained blank, I then repeated, Just a blank and nothing more.
                </p>,
                <p
                  className="mt20"
                >
                  With a scream, I was defeated, For my cookies were deleted, So i begged, no longer seated, "Give me back my hrefs once more!"
                </p>,
                <p
                  className="mt20"
                >
                  Then, in answer to my query, Through the net I loved so dearly, Came its answer, dark and dreary: Quoth the server, 404
                </p>,
              ],
              "className": "card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Oh, Hello 404",
                },
                "ref": null,
                "rendered": "Oh, Hello 404",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Whatever you were trying to do, it didn't work",
                  "className": "mt10",
                },
                "ref": null,
                "rendered": "Whatever you were trying to do, it didn't work",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Once upon a midnight dreary, While I websurfed, weak and weary, Over many a strange and spurious websites or blogs galore,",
                  "className": "mt20",
                },
                "ref": null,
                "rendered": "Once upon a midnight dreary, While I websurfed, weak and weary, Over many a strange and spurious websites or blogs galore,",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "While I clicked my fav'rite bookmark, Suddenly there came a warning, And my heart was filled with mourning, Mourning for my dear amour.",
                  "className": "mt20",
                },
                "ref": null,
                "rendered": "While I clicked my fav'rite bookmark, Suddenly there came a warning, And my heart was filled with mourning, Mourning for my dear amour.",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "'Tis not possible!, I pleaded, But my browser, so conceited, Remained blank, I then repeated, Just a blank and nothing more.",
                  "className": "mt20",
                },
                "ref": null,
                "rendered": "'Tis not possible!, I pleaded, But my browser, so conceited, Remained blank, I then repeated, Just a blank and nothing more.",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "With a scream, I was defeated, For my cookies were deleted, So i begged, no longer seated, \\"Give me back my hrefs once more!\\"",
                  "className": "mt20",
                },
                "ref": null,
                "rendered": "With a scream, I was defeated, For my cookies were deleted, So i begged, no longer seated, \\"Give me back my hrefs once more!\\"",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Then, in answer to my query, Through the net I loved so dearly, Came its answer, dark and dreary: Quoth the server, 404",
                  "className": "mt20",
                },
                "ref": null,
                "rendered": "Then, in answer to my query, Through the net I loved so dearly, Came its answer, dark and dreary: Quoth the server, 404",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
